### DISCUSSION TITLE
Asterisk in args.input.dir

### DISCUSSION THREAD

**Discussion by kendalldita** [dita-ot/dita-ot#4503] [Q&A] [answered]
I've inherited code, modified dita-ot 3.6.1, that passes --args.input.dir=path/* -i bookmap.ditamap. Somehow that works, for some publications, but not if I use out of the box dita-ot 3.6.1. Do you have any idea what the purpose of that could be? I'm asked not to change the parameters.

**Comment by jelovirt** [comment, marked as answer]
I wouldn't expect that to work 🤔 

If it was `--args.input.dir path/*`, then the shell would expand that to `path/*` whatever is in `path` directory, e.g. `path/product` if `product` was the only directory in `path`. Then if `path/subdir/bookmap.ditamap` would be used as the input resource. But not sure if that would work either.

You could debug this by changing modifying https://github.com/dita-ot/dita-ot/blob/3.6.1/src/main/bin/dita#L179 to use `echo` instead of `eval`. Maybe that shows you what's being passed to the DITA process.

**Reply by kendalldita** [reply]
Thanks. `echo` just showed that the asterisk is there. Because I'm required to keep the parameters that way, and I need to use dita.bat as well, I modified Main.java to remove the asterisk. I have to use dita.bat also. I will argue for removing the asterisk and also upgrading ot.
